### PR TITLE
Test Data Generation - only show necessary columns of tables used in query

### PR DESCRIPTION
**Pull request by YannanGao-gs**
#### What does this PR do / why is it needed ?

Test Data Generation - only show necessary columns (**not null | primary key | used columns**) of tables used in query

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

**Review by MauricioUyaguari — DISMISSED** [review]

**Review by akphi — APPROVED** [review]

File: legend-engine-core/legend-engine-core-test/legend-engine-test-data-generation/src/main/java/org/finos/legend/engine/testData/generation/service/TestDataGenerationService.java [hunk 1]
```diff
@@ -16,42 +16,39 @@
...
 {
-    public static List<EmbeddedData> generateEmbeddedData(Lambda query, Root_meta_pure_runtime_Runtime runtime, Mapping mapping, PureModel pureModel)
+    public static List<EmbeddedData> generateEmbeddedData(Lambda query, Mapping mapping, PureModel pureModel)
     {
-        RichIterable<? extends Table> tables = core_relational_relational_testDataGeneration_testDataGeneration.Root_meta_relational_testDataGeneration_getTableFromQuery_FunctionDefinition_1__Mapping_1__Runtime_1__Table_MANY_(
-                buildPureLambda(query, pureModel), mapping, runtime, pureModel.getExecutionSupport());
-        if (tables.isEmpty())
+        Root_meta_relational_metamodel_data_RelationalCSVData relationalCSVData = core_relational_relational_testDataGeneration_testDataGeneration.Root_meta_relational_testDataGeneration_getRelationalCSVDataFromQuery_FunctionDefinition_1__Mapping_1__RelationalCSVData_1_(
+                buildPureLambda(query, pureModel), mapping, pureModel.getExecutionSupport());
+        if (relationalCSVData == null)
```
Comment (davidharte-gs) [inline legend-engine-core/legend-engine-core-test/legend-engine-test-data-generation/src/main/java/org/finos/legend/engine/testData/generation/service/TestDataGenerationService.java:39]: is it even possible to get Null back? seems like an anti-pattern

File: legend-engine-core/legend-engine-core-test/legend-engine-test-data-generation/src/main/java/org/finos/legend/engine/testData/generation/service/TestDataGenerationService.java [hunk 2]
```diff
@@ -16,42 +16,39 @@
...
         {
             return null;
         }
-        List<RelationalCSVTable> relationalCSVTables = tables.collect(table ->
+        RelationalCSVData data = new RelationalCSVData();
```
Comment (davidharte-gs) [inline legend-engine-core/legend-engine-core-test/legend-engine-test-data-generation/src/main/java/org/finos/legend/engine/testData/generation/service/TestDataGenerationService.java:43]: shouldn't we be calling transformEmbeddedData to transform the results to the Pure protocol and then sending that back?

File: legend-engine-core/legend-engine-core-test/legend-engine-test-data-generation/src/main/java/org/finos/legend/engine/testData/generation/service/TestDataGenerationService.java [hunk 3]
```diff
@@ -16,42 +16,39 @@
...
 import org.finos.legend.pure.m3.coreinstance.meta.pure.mapping.Mapping;
 import org.finos.legend.pure.generated.core_relational_relational_testDataGeneration_testDataGeneration;
 import org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.function.LambdaFunction;
-import org.finos.legend.pure.m3.coreinstance.meta.relational.metamodel.Column;
-import org.finos.legend.pure.m3.coreinstance.meta.relational.metamodel.relation.Table;
 
 import java.util.Collections;
 import java.util.List;
 
 public class TestDataGenerationService
```
Comment (davidharte-gs) [inline legend-engine-core/legend-engine-core-test/legend-engine-test-data-generation/src/main/java/org/finos/legend/engine/testData/generation/service/TestDataGenerationService.java:33]: should this be renamed to TestDataBuilder, it's not really a 'Service', actually the class 'TestDataGeneration' looks more like the Service

File: legend-engine-core/legend-engine-core-test/legend-engine-test-data-generation/pom.xml [hunk 4]
```diff
@@ -16,10 +16,6 @@
             <groupId>org.finos.legend.pure</groupId>
             <artifactId>legend-pure-m3-core</artifactId>
         </dependency>
-        <dependency>
```
Comment (davidharte-gs) [inline legend-engine-core/legend-engine-core-test/legend-engine-test-data-generation/pom.xml:19]: no longer needed?

**Review by davidharte-gs — CHANGES_REQUESTED** [review]

File: legend-engine-core/legend-engine-core-test/legend-engine-test-data-generation/pom.xml (hunk 4 above)
Comment (YannanGao-gs) [reply legend-engine-core/legend-engine-core-test/legend-engine-test-data-generation/pom.xml:19]: yes, it's not needed.

File: legend-engine-core/legend-engine-core-test/legend-engine-test-data-generation/src/main/java/org/finos/legend/engine/testData/generation/service/TestDataGenerationService.java (hunk 3 above)
Comment (YannanGao-gs) [reply legend-engine-core/legend-engine-core-test/legend-engine-test-data-generation/src/main/java/org/finos/legend/engine/testData/generation/service/TestDataGenerationService.java:33]: As discussed with Mauricio, leave this comment open as it requires corresponding changes in alloy.

File: legend-engine-core/legend-engine-core-test/legend-engine-test-data-generation/src/main/java/org/finos/legend/engine/testData/generation/service/TestDataGenerationService.java (hunk 2 above)
Comment (YannanGao-gs) [reply legend-engine-core/legend-engine-core-test/legend-engine-test-data-generation/src/main/java/org/finos/legend/engine/testData/generation/service/TestDataGenerationService.java:39]: May I know where `transformEmbeddedData` is defined? I haven't found any transform functions in the protocol yet.

Comment (davidharte-gs) [reply legend-engine-core/legend-engine-core-test/legend-engine-test-data-generation/src/main/java/org/finos/legend/engine/testData/generation/service/TestDataGenerationService.java:39]: meta::protocols::pure::vX_X_X::transformation::fromPureGraph::runtime::transformEmbeddedData

**Review by davidharte-gs — CHANGES_REQUESTED** [review]

File: legend-engine-core/legend-engine-core-test/legend-engine-test-data-generation/src/main/java/org/finos/legend/engine/testData/generation/service/TestDataGenerationService.java (hunk 2 above)
Comment (YannanGao-gs) [reply legend-engine-core/legend-engine-core-test/legend-engine-test-data-generation/src/main/java/org/finos/legend/engine/testData/generation/service/TestDataGenerationService.java:39]: Thanks. but..This function is not open sourced yet.

**Review by davidharte-gs — APPROVED** [review]

**Pull request merged**